### PR TITLE
Fix: Ensure GIF and TIFF are treated as images

### LINKED ISSUE #12133: Bug: GIF and TIFF aren't treated as images
### Description

GIF and TIFF aren't treated as images in Files, so they can't be set as the background or rotated.

### Steps To Reproduce

1. Select a GIF or TIFF image.
2. "Set as background" and "Rotate" option isn't appear.

### Requirements

Ensure GIF and TIFF are treated as images, as well as BMP, JPEG, PNG.

### Files Version

2.4.66.0

### Windows Version

Windows 11 22H2

### Log file

n/a

### PR DESCRIPTION
**Pull request by f-pereira**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #12133

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Open app
   2. Select a .gif or .tiff file
   3. Should have image options (rotate left, rotate right, etc...)

**Screenshots (optional)**
Screenshot 2023-04-22 163710

### CODE REVIEW

File: src/Files.Backend/Helpers/FileExtensionHelpers.cs [hunk 1]
```diff
@@ -26,7 +26,7 @@ public static bool HasExtension(string? filePathToCheck, params string[] extensi
 		/// <returns><c>true</c> if the fileExtensionToCheck is an image;
 		/// otherwise, <c>false</c>.</returns>
 		public static bool IsImageFile(string? fileExtensionToCheck)
-			=> HasExtension(fileExtensionToCheck, ".png", ".bmp", ".jpg", ".jpeg");
+			=> HasExtension(fileExtensionToCheck, ".png", ".bmp", ".jpg", ".jpeg", ".gif", ".tiff");
```
Comment (ferrariofilippo) [inline src/Files.Backend/Helpers/FileExtensionHelpers.cs:29]: You should also include `.tif`, since it is the same as `.tiff`

Comment (f-pereira) [reply src/Files.Backend/Helpers/FileExtensionHelpers.cs]: You're absolutely correct, my bad!

**Comment by ferrariofilippo** [comment]
There's a bug when rotating GIFs, but I think it's a different issue

**Comment by f-pereira** [comment]
Oh I see, It seems that only the first frame is rotated

**Comment by hishitetsu** [comment]
Thank you for your PR! This page will help you to rotate all frames.
https://learn.microsoft.com/en-us/uwp/api/windows.graphics.imaging.bitmapencoder?view=winrt-22621

> If you want to encode an additional frame, call GoToNextFrameAsync.

**Comment by f-pereira** [comment]
@hishitetsu Thank you for the documentation! I tried to use it and I found a problem that I am Looking for solution: the first frame rotates well, but if the next frame have a different width/height, it rotates well but It is misplaced (e.g It is placed on top right corner instead of center)

**Comment by hishitetsu** [comment]
* Will there be any problem if all frames had the same width and height?
* Will rotating four times in a row or rotating left and then right restore all frames to there original state even if there are frames of different widths and heights?

If no to the former and yes to the latter, I think that is still fine.

**Comment by f-pereira** [comment]
@hishitetsu Done! 
I tested with a gif where all the frames are equal in size and it worked well.
I also tested with other image formats and it is working

**Comment by ferrariofilippo** [comment]
> @hishitetsu Done! I tested with a gif where all the frames are equal in size and it worked well. I also tested with other image formats and it is working

I'm getting a frame wrong (but if I rotate the gif 4 times everything works). I used <a href="https://upload.wikimedia.org/wikipedia/commons/thumb/2/2c/Rotating_earth_%28large%29.gif/260px-Rotating_earth_%28large%29.gif">this</a> to test

**Comment by f-pereira** [comment]
I noticed that it happens when the frames are not equal in size (I checked your gif, the frames size are different)
To confirm this hypothesis I used this gif

 this

File: src/Files.App/Helpers/BitmapHelper.cs [hunk 2]
```diff
@@ -63,6 +63,12 @@ public static async Task Rotate(string filePath, BitmapRotation rotation)
 			using var memStream = new InMemoryRandomAccessStream();
 			BitmapEncoder encoder = await BitmapEncoder.CreateForTranscodingAsync(memStream, decoder);
 
+			for(int i = 0; i < decoder.FrameCount - 1; i++)
```
Comment (hishitetsu) [inline src/Files.App/Helpers/BitmapHelper.cs:66]:
```suggestion
			for (int i = 0; i < decoder.FrameCount - 1; i++) 
```

**Review by ferrariofilippo — APPROVED** [review]
LGTM and works for me!

**Review by yaira2 — APPROVED** [review]
Thank you

**Pull request merged**